### ISSUE TITLE
Fix filters boolean logic

### ISSUE THREAD

**Issue by phoebus-84** [interfacerproject/zenflows#38]
On interfacer-gui we have a filters like this for resources: {notCustodian: id, orName: string, orNote: string}. It returns resources that respect the condition: notCoustodian or Name or Note. It would be useful to have something to express the statement: notCustodian and (name or note).

**Comment by srfsh** [comment]
Yeah, it is sorta hard to get right.  Those `or*` and `not*` prefixes were a quick workaround.  I'll study and see what we can do about it.

At worst, we'll use a custom string format for the filters and parse it.
